File: TAL.Developer.Test.Domain/Models/TimesheetsReportModel.cs
using System;

namespace TAL.Developer.Test.Domain.Models
{
    public class TimesheetsReportModel
    {
        public int EmployeeId { get; set; }
        public string EmployeeName { get; set; }
        public string EmployeeUsername { get; set; }
        public string TimezoneName { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal Rate { get; set; }
        public string Status { get; set; }
        public decimal? HoursWorked { get; set; }
        public decimal? Cost { get; set; }
    }
}

File: TAL.Developer.Test.Site/ClientApp/src/app/models/TimesheetsReport.model.ts
export interface TimesheetsReport {
  employeeId: number;
  employeeName: string;
  employeeUsername: string;
  timezoneName: string;
  startDate: Date;
  formattedStartDate: string;
  endDate: Date;
  formattedEndDate: string;
  rate: number;
  status: string;
  hoursWorked: number;
  cost: number;
}
